data/public_data/transformer.py

```diff
@@ -142,68 +143,129 @@ def parse_cinema_object_to_data(self, cinema_object):
-                title = title.replace("[IMAX]", "")
-                additional_info.append("IMAX")
-            if "[M]" in title:
-                title = title.replace("[M]", "")
-            if "[IMAX 3D]" in title:
-                title = title.replace("[IMAX 3D]", "")
-                additional_info.append("IMAX")
-                additional_info.append("3D")
-
-        else:
-            raise Exception("Invalid cinema provider")
+        # special rules
+        if "Kungfu" in title:
+            title = title.replace("Kungfu", "Kung-fu")
+
+        # general rules
+        if "`" in title:
+            title = title.replace("`", "\'")
+        if "[D]" in title:
+            title = title.replace("[D]", "")
+            additional_info.append("Digital")
+        if "[IMAX]" in title:
+            title = title.replace("[IMAX]", "")
+            additional_info.append("IMAX")
+        if "[M]" in title:
+            title = title.replace("[M]", "")
+        if "[IMAX 3D]" in title:
+            title = title.replace("[IMAX 3D]", "")
+            additional_info.append("IMAX")
+            additional_info.append("3D")
 
+        return self._parse_title_and_info(title, additional_info)
+
+    def _parse_cathay_movie_title(self, title):
+        """
+        parse the raw title displayed on cathay website into
+        clean movie title plus a list of movie types
+        :param title: string
+        :return: string, list
+        """
+        additional_info = []
+        if "*" in title:
+            title = title.replace("*", "")
+            # have not figure out the meaning of *
+        if "(Dolby Digital)" in title:
+            tokens = title.split(" ")
+            splitter = tokens.index("(Dolby")
+            title = " ".join(tokens[:splitter - 1])
+            additional_info.append("Dolby Digital")
+        if "(Dolby Atmos)" in title:
+            tokens = title.split(" ")
+            splitter = tokens.index("(Dolby")
+            title = " ".join(tokens[:splitter - 1])
+            additional_info.append("Dolby Atmos")
+            title = title.replace("Atmos", "")
+
+        return self._parse_title_and_info(title, additional_info)
+
+    @staticmethod
+    def _parse_title_and_info(title, additional_info):
+        """
+        further clean the title and the additional information,
+        ready to be stored in database
+        :param title: string
+        :param info: list
+        :return: string, string
+        """
         title = title.strip()
-        additional_info = ",".join(additional_info)
-        return title, additional_info
+        info = ", ".join(additional_info)
+        return title, info
+
+
+class CinemaListTransformer:
+
+    @staticmethod
+    def insert_cinema_data(cinema_name, cinema_url, provider):
+        inserted_tuple = {
+            "url": cinema_url,
+            "cinema_name": cinema_name,
+            "provider": provider
+        }
+        return inserted_tuple
```

Comment:
oh u mean this, i don't like directly return, looks ugly when you have a lot of things on the return statement